test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,171 @@
+                std::to_string(((WT_CONNECTION_IMPL *)conn)->cache_size));
+
+            /*
+             * We can only track the operation when the cache size is big enough otherwise it will
+             * be rolled back.
+             */
+            if (increase_cache) {
+                /*
+                 * These dummy fields are required by the save_operation API but not needed for this
+                 * test.
+                 */
+                const uint64_t collection_id = 0;
+                const std::string key = "";
+                const std::string value = "";
+                tc->transaction.begin();
+                testutil_check(tc->tracking->save_operation(tracking_operation::CUSTOM,
+                  collection_id, key, value, tc->tsm->get_next_ts(), tc->op_track_cursor));
+                testutil_assert(tc->transaction.commit());
+            }
+            increase_cache = !increase_cache;
+        }
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        uint64_t coll_id = 0;
+        collection &coll = tc->db.get_collection(coll_id);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+
+        while (tc->running()) {
+
+            tc->transaction.try_begin();
+
+            /* Insert a random value. */
+            const std::string value =
+              random_generator::instance().generate_pseudo_random_string(tc->value_size);
+            bool ret = tc->insert(cursor, coll.id, value);
+
+            if (!ret) {
+                tc->transaction.rollback();
+            } else if (tc->transaction.can_commit()) {
+                /*
+                 * The transaction can fit in the current cache size and is ready to be committed.
+                 * This means the tracking table will contain a new record to represent this
+                 * transaction which will be used during the validation stage.
+                 */
+                testutil_assert(tc->transaction.commit());
+            }
+
+            tc->sleep();
+        }
+
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        if (tc->transaction.active())
+            tc->transaction.rollback();
+    }
+
+    void
+    validate(const std::string &operation_table_name, const std::string &,
+      const std::vector<uint64_t> &) override final
+    {
+        WT_DECL_RET;
+        int tracked_op_type;
+        tracking_operation op_type;
+        uint64_t num_records = 0, cache_size;
+
+        /* Open a cursor on the tracking table to read it. */
+        scoped_session session = connection_manager::instance().create_session();
+        scoped_cursor cursor = session.open_scoped_cursor(operation_table_name);
+
+        /* Parse the tracking table. */
+        while ((ret = cursor->next(cursor.get())) == 0) {
+            testutil_check(cursor->get_value(cursor.get(), &tracked_op_type, &cache_size));
```

Comment:
`cache_size` isn't saved when writing `op_type == INSERT` records into the tracking table, but we're reading it here. Either WT is smart enough not to read out empty fields or we should add some defensive checks to only read `cache_size` when `op_type == CUSTOM`

Reply:
The cache_size is always written whatever the operation is, see [here](https://github.com/wiredtiger/wiredtiger/pull/7902/files#diff-e12c1ae126af9008c48745f0d1603ef8ad756233627ccc445aa530a801c00b01R49).